timesketch/api/v1/resources/export.py

```diff
@@ -0,0 +1,175 @@
+
+        form = forms.ExploreForm.build(request)
+        if not form.validate_on_submit():
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Unable to export data, unable to validate form data",
+            )
+
+        query_dsl = form.dsl.data
+        query_filter = request.json.get("filter") or {}
+        return_field_string = form.fields.data
+
+        # Resolve return fields
+        if return_field_string:
+            return_fields = [x.strip() for x in return_field_string.split(",")]
+        else:
+            return_fields = query_filter.get("fields", [])
+            return_fields = [field["field"] for field in return_fields]
+            if not return_fields:
+                return_fields = DEFAULT_SOURCE_FIELDS
+
+        # Resolve Indices
+        all_timeline_ids = [t.id for t in sketch.timelines]
+        indices = query_filter.get("indices", all_timeline_ids)
+        if "_all" in indices:
+            indices = all_timeline_ids
+
+        # Validate indices and get timeline IDs (checks permissions implicitly)
+        indices, timeline_ids = utils.get_validated_indices(indices, sketch)
+
+        # Map to actual OpenSearch index names for the PIT
+        indices_for_pit = []
+        for timeline in sketch.timelines:
+            if timeline.id in timeline_ids:
+                indices_for_pit.append(timeline.searchindex.index_name)
+
+        # Final check to ensure indices exist in backend
+        indices_for_pit = utils.validate_indices(indices_for_pit, self.datastore)
+
+        if not indices_for_pit:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "No valid search indices were found to perform the export on.",
+            )
+
+        # Build the base query DSL using standard logic to ensure timeline isolation
+        full_query_dsl = self.datastore.build_query(
+            sketch.id,
+            form.query.data,
+            query_filter,
+            query_dsl,
+            None,  # No aggregations
+            timeline_ids,
+        )
+
+        # Prepare the lightweight body for slicing.
+        # Slicing handles sort/size/pagination, so we only extract query/filter/source.
+        base_query_body = {
+            "query": full_query_dsl.get("query", {}),
+        }
+
+        if return_fields:
+            base_query_body["_source"] = return_fields
+
+        if "post_filter" in full_query_dsl:
+            base_query_body["post_filter"] = full_query_dsl["post_filter"]
+
+        def generate():
+            try:
+                event_generator = self.datastore.export_events_with_slicing(
+                    indices_for_pit=indices_for_pit,
+                    base_query_body=base_query_body,
+                )
+
+                for event in event_generator:
+                    yield json.dumps(event) + "\n"
+            except Exception as e:
+                logger.error("Error during streaming export: %s", e, exc_info=True)
```

Comment:
please add some more information to it that would help in troubleshooting sessions like sketch_id, timeline_id ....